components/ai_chat_ui/components/conversation_list/index.tsx

```diff
@@ -16,6 +16,19 @@ interface ConversationListProps {
 }
 
 function ConversationList (props: ConversationListProps) {
+  // Scroll the last conversation item in to view when entries are added.
+  React.useEffect(() => {
+    if (!props.list.length) {
+      return
+    }
+    requestAnimationFrame(() => {
+      document.querySelector(`[data-conversation-entry-id="${props.list.length - 1}"]`)?.scrollIntoView({
```

Comment:
querySelector will have to walk through the nodes to find, whereas the ref handle will be attached as soon as the node gets created. It's O(n) vs O(1). There are other nuances, but they don't apply to our use-case. Prefer using refs here.